### ISSUE TITLE
SetAudioOutputTest does not work as expected (not similar to vlc player)

### ISSUE THREAD

**Issue by Joe-Merten** [caprica/vlcj#303]
Hey Mark,

Tested on both kubuntu 14.04 32 bit and 64 bit using vlcj git rev 20e63beece758dc23695e34f65114bc0fa3330da.

My system has 2 audio outputs.
- Internal Audio Analog Stereo (Onboard Audio, primary device)
- BM-8810 (Bluetooth Headphone, secondary device)

When using e.g. vlc player, I can see them both within the "Audio" menu:
![vlc player audio outputs](https://cloud.githubusercontent.com/assets/6552470/5897654/ba6c3a7c-a546-11e4-878f-809eb3e3ef59.png)
Changing the audio output during playback switches the signal immediatly between the two outputs.
This all works as expected.

Now I try to provide the same functionality in my vlcj based application.

I'd took a look to your SetAudioOutputTest, but the behavior looks quite different.
I could not see strings like "Internal Audio Analog Stereo" or "BM-8810" (see attached screenshot).
![vlcj setaudiooutputtest](https://cloud.githubusercontent.com/assets/6552470/5897662/d4786e90-a546-11e4-8abf-1374a3c4da6a.png)

Any idea how can I determine this two strings which I see on vlc player and how to switch the output?

I'd tried all alternatives which were provided by SetAudioOutputTest with the following results:

- Selecting "Pulseaudio audio output" directs the audio signal to the audio device, which I recently have selected in the vlc player.
- Alsa audio output / default -> output to primary device (Internal Audio Analog Stereo)
- Alsa audio output / null -> dito
- Alsa audio output / pulse -> no output
- Alsa audio output / sysdefault:CARD=Intel -> output to primary device
- Alsa audio output / front:CARD=Intel -> output to primary device
- Alsa audio output / surround40:CARD=Intel -> output to primary device
- Alsa audio output / surround41:CARD=Intel -> output to primary device
- Alsa audio output / surround50:CARD=Intel -> libvlc errors: "cannot disable resampling", "module not functional", "failed to create audio output"
- Alsa audio output / surround51:CARD=Intel -> libvlc errors as above
- Alsa audio output / surround71:CARD=Intel -> libvlc errors as above
- Alsa audio output / dmix:CARD=INTEL -> output to primary device
- Alsa audio output / dsnoop:CARD=INTEL -> output to primary device
- Alsa audio output / hw:CARD=INTEL -> libvlc errors similar as above
- Alsa audio output / plughw:CARD=INTEL -> output to primary device

Further, in vlc player, I see a difference if the bluetooth headset is connected or not (so I see either one or two entries in the menu).
In SetAudioOutputTest, there are no differences in the menu entries, regardless if the bluetooth headset is connected or not.

And next thing is that I always have to restart the playback to apply the new audio output selection. May there be a way to switch the output during playback?

Logging output of vlcj SetAudioOutputTest (startup, play, close):

    $ java -Dvlcj.log=DEBUG -cp vlcj-3.2.0-SNAPSHOT-tests.jar uk.co.caprica.vlcj.test.factory.SetAudioOutputTest /D/MP3/OGG-WMA-RM-Test/Testfiles/America\ -\ The\ Last\ Unicorn.mp3 
    vlcj: (MediaPlayerFactory.java:137)                  | DEBUG | Trying workaround for Java7 on Linux
    vlcj: (MediaPlayerFactory.java:143)                  | DEBUG | Attempting to load jawt...
    vlcj: (MediaPlayerFactory.java:145)                  | DEBUG | ...loaded jawt
    vlcj: (MediaPlayerFactory.java:153)                  | DEBUG | Java7 on Linux workaround complete.
    vlcj: (Info.java:70)                                 | INFO  | vlcj: 3.2.0-SNAPSHOT
    vlcj: (Info.java:71)                                 | INFO  | java: 1.7.0_65 Oracle Corporation
    vlcj: (Info.java:72)                                 | INFO  | java home: /usr/lib/jvm/java-7-openjdk-i386/jre
    vlcj: (Info.java:73)                                 | INFO  | os: Linux 3.13.0-44-generic i386
    vlcj: (LibVlcFactory.java:164)                       | INFO  | vlc: 2.2.0-rc2 Weatherwax, changeset 2.2.0-rc1-118-g22fda39
    vlcj: (LibVlcFactory.java:165)                       | INFO  | libvlc: libvlc.so
    vlcj: (MediaPlayerFactory.java:261)                  | DEBUG | MediaPlayerFactory(libvlc=Proxy interface to Native Library <libvlc.so@1756576752>,libvlcArgs=[])
    vlcj: (MediaPlayerFactory.java:263)                  | DEBUG | jna.library.path=null
    vlcj: (MediaPlayerFactory.java:272)                  | DEBUG | VLC_PLUGIN_PATH=null
    vlcj: (MediaPlayerFactory.java:275)                  | DEBUG | instance=native@0x68b63090 (uk.co.caprica.vlcj.binding.internal.libvlc_instance_t@68b63090)
    vlcj: (MediaPlayerFactory.java:283)                  | DEBUG | equalizerAvailable=true
    vlcj: (MediaPlayerFactory.java:506)                  | DEBUG | createEqualizerBandFrequencies()
    vlcj: (MediaPlayerFactory.java:508)                  | DEBUG | numBands=10
    vlcj: (MediaPlayerFactory.java:513)                  | DEBUG | result=[31.25, 62.5, 125.0, 250.0, 500.0, 1000.0, 2000.0, 4000.0, 8000.0, 16000.0]
    vlcj: (MediaPlayerFactory.java:523)                  | DEBUG | createEqualizerPresetNames()
    vlcj: (MediaPlayerFactory.java:525)                  | DEBUG | numPresets=18
    vlcj: (MediaPlayerFactory.java:530)                  | DEBUG | result=[Flat, Classical, Club, Dance, Full bass, Full bass and treble, Full treble, Headphones, Large Hall, Live, Party, Pop, Reggae, Rock, Ska, Soft, Soft rock, Techno]
    vlcj: (MediaPlayerFactory.java:645)                  | DEBUG | newEmbeddedMediaPlayer()
    vlcj: (MediaPlayerFactory.java:656)                  | DEBUG | newEmbeddedMediaPlayer(fullScreenStrategy=null)
    vlcj: (DefaultMediaPlayer.java:190)                  | DEBUG | DefaultMediaPlayer(libvlc=Proxy interface to Native Library <libvlc.so@1756576752>, instance=native@0x68b63090 (uk.co.caprica.vlcj.binding.internal.libvlc_instance_t@68b63090))
    vlcj: (DefaultMediaPlayer.java:1697)                 | DEBUG | createInstance()
    vlcj: (DefaultMediaPlayer.java:1700)                 | DEBUG | mediaPlayerInstance=native@0x68b77d80 (uk.co.caprica.vlcj.binding.internal.libvlc_media_player_t@68b77d80)
    vlcj: (DefaultMediaPlayer.java:1703)                 | DEBUG | mediaPlayerEventManager=native@0x68604ae8 (uk.co.caprica.vlcj.binding.internal.libvlc_event_manager_t@68604ae8)
    vlcj: (DefaultMediaPlayer.java:1760)                 | DEBUG | registerEventListener()
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerMediaChanged
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerNothingSpecial
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerOpening
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerBuffering
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerPlaying
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerPaused
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerStopped
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerForward
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerBackward
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerEndReached
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerEncounteredError
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerTimeChanged
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerPositionChanged
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerSeekableChanged
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerPausableChanged
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerTitleChanged
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerSnapshotTaken
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerLengthChanged
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerVout
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerScrambledChanged
    vlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerESAdded
    This object event manager doesn't know about 'Unknown Event' eventsvlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=-1
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerESDeleted
    This object event manager doesn't know about 'Unknown Event' eventsvlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=-1
    vlcj: (DefaultMediaPlayer.java:1765)                 | DEBUG | event=libvlc_MediaPlayerESSelected
    This object event manager doesn't know about 'Unknown Event' eventsvlcj: (DefaultMediaPlayer.java:1767)                 | DEBUG | result=-1
    vlcj: (MediaPlayerFactory.java:380)                  | DEBUG | getAudioOutputs()
    vlcj: (MediaPlayerFactory.java:409)                  | DEBUG | getAudioOutputDevices(outputName=pulse)
    vlcj: (MediaPlayerFactory.java:409)                  | DEBUG | getAudioOutputDevices(outputName=adummy)
    vlcj: (MediaPlayerFactory.java:409)                  | DEBUG | getAudioOutputDevices(outputName=alsa)
    vlcj: (MediaPlayerFactory.java:409)                  | DEBUG | getAudioOutputDevices(outputName=amem)
    vlcj: (MediaPlayerFactory.java:409)                  | DEBUG | getAudioOutputDevices(outputName=afile)
    vlcj: (MediaPlayerFactory.java:719)                  | DEBUG | newVideoSurface(canvas=java.awt.Canvas[canvas0,0,0,0x0,invalid])
    vlcj: (MediaPlayerFactory.java:734)                  | DEBUG | videoSurface=uk.co.caprica.vlcj.player.embedded.videosurface.CanvasVideoSurface@c07b39
    vlcj: (DefaultEmbeddedMediaPlayer.java:142)          | DEBUG | setVideoSurface(videoSurface=uk.co.caprica.vlcj.player.embedded.videosurface.CanvasVideoSurface@c07b39)
    vlcj: (DefaultMediaPlayer.java:222)                  | DEBUG | playMedia(mrl=/D/MP3/OGG-WMA-RM-Test/Testfiles/America - The Last Unicorn.mp3,mediaOptions=[])
    vlcj: (DefaultMediaPlayer.java:236)                  | DEBUG | prepareMedia(mrl=/D/MP3/OGG-WMA-RM-Test/Testfiles/America - The Last Unicorn.mp3,mediaOptions=[])
    vlcj: (DefaultMediaPlayer.java:1892)                 | DEBUG | setMedia(media=/D/MP3/OGG-WMA-RM-Test/Testfiles/America - The Last Unicorn.mp3,mediaOptions=[])
    vlcj: (MediaResourceLocator.java:83)                 | DEBUG | encodeMrl(mrl=/D/MP3/OGG-WMA-RM-Test/Testfiles/America - The Last Unicorn.mp3)
    vlcj: (MediaResourceLocator.java:110)                | DEBUG | MRL does not contain any Unicode characters
    vlcj: (MediaResourceLocator.java:112)                | DEBUG | result=/D/MP3/OGG-WMA-RM-Test/Testfiles/America - The Last Unicorn.mp3
    vlcj: (DefaultMediaPlayer.java:1911)                 | DEBUG | Treating mrl as a path
    vlcj: (DefaultMediaPlayer.java:1914)                 | DEBUG | mediaInstance=native@0x686db828 (uk.co.caprica.vlcj.binding.internal.libvlc_media_t@686db828)
    vlcj: (DefaultMediaPlayer.java:1816)                 | DEBUG | registerMediaEventListener()
    vlcj: (DefaultMediaPlayer.java:1823)                 | DEBUG | event=libvlc_MediaMetaChanged
    vlcj: (DefaultMediaPlayer.java:1825)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1823)                 | DEBUG | event=libvlc_MediaSubItemAdded
    vlcj: (DefaultMediaPlayer.java:1825)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1823)                 | DEBUG | event=libvlc_MediaDurationChanged
    vlcj: (DefaultMediaPlayer.java:1825)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1823)                 | DEBUG | event=libvlc_MediaParsedChanged
    vlcj: (DefaultMediaPlayer.java:1825)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1823)                 | DEBUG | event=libvlc_MediaFreed
    vlcj: (DefaultMediaPlayer.java:1825)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1823)                 | DEBUG | event=libvlc_MediaStateChanged
    vlcj: (DefaultMediaPlayer.java:1825)                 | DEBUG | result=0
    vlcj: (DefaultMediaPlayer.java:1649)                 | DEBUG | mrl(mediaInstance=native@0x686db828 (uk.co.caprica.vlcj.binding.internal.libvlc_media_t@686db828))
    vlcj: (DefaultMediaPlayer.java:714)                  | DEBUG | play()
    vlcj: (DefaultEmbeddedMediaPlayer.java:314)          | DEBUG | onBeforePlay()
    vlcj: (DefaultEmbeddedMediaPlayer.java:151)          | DEBUG | attachVideoSurface()
    vlcj: (CanvasVideoSurface.java:67)                   | DEBUG | attach()
    vlcj: (DefaultMediaPlayer.java:2061)                 | DEBUG | mediaChanged(mediaPlayer=uk.co.caprica.vlcj.player.embedded.DefaultEmbeddedMediaPlayer@f3c7ad,media=native@0x686db828 (uk.co.caprica.vlcj.binding.internal.libvlc_media_t@686db828),mrl=file:///D/MP3/OGG-WMA-RM-Test/Testfiles/America%20-%20The%20Last%20Unicorn.mp3)
    vlcj: (DefaultMediaPlayer.java:2065)                 | DEBUG | Raising event for new media
    vlcj: (CanvasVideoSurface.java:70)                   | DEBUG | componentId=77594638
    vlcj: (LinuxVideoSurfaceAdapter.java:39)             | DEBUG | attach(componentId=77594638)
    vlcj: (CanvasVideoSurface.java:72)                   | DEBUG | video surface attached
    vlcj: (DefaultMediaPlayer.java:717)                  | DEBUG | after play
    vlcj: (DefaultMediaPlayer.java:1667)                 | DEBUG | release()
    vlcj: (DefaultMediaPlayer.java:1717)                 | DEBUG | destroyInstance()
    vlcj: (DefaultMediaPlayer.java:1719)                 | DEBUG | Detach media events...
    vlcj: (DefaultMediaPlayer.java:1835)                 | DEBUG | deregisterMediaEventListener()
    vlcj: (DefaultMediaPlayer.java:1842)                 | DEBUG | event=libvlc_MediaMetaChanged
    vlcj: (DefaultMediaPlayer.java:1842)                 | DEBUG | event=libvlc_MediaSubItemAdded
    vlcj: (DefaultMediaPlayer.java:1842)                 | DEBUG | event=libvlc_MediaDurationChanged
    vlcj: (DefaultMediaPlayer.java:1842)                 | DEBUG | event=libvlc_MediaParsedChanged
    vlcj: (DefaultMediaPlayer.java:1842)                 | DEBUG | event=libvlc_MediaFreed
    vlcj: (DefaultMediaPlayer.java:1842)                 | DEBUG | event=libvlc_MediaStateChanged
    vlcj: (DefaultMediaPlayer.java:1721)                 | DEBUG | Media events detached.
    vlcj: (DefaultMediaPlayer.java:1724)                 | DEBUG | Release media...
    vlcj: (DefaultMediaPlayer.java:1726)                 | DEBUG | Media released.
    vlcj: (DefaultMediaPlayer.java:1729)                 | DEBUG | Detach media player events...
    vlcj: (DefaultMediaPlayer.java:1779)                 | DEBUG | deregisterEventListener()
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerMediaChanged
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerNothingSpecial
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerOpening
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerBuffering
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerPlaying
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerPaused
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerStopped
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerForward
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerBackward
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerEndReached
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerEncounteredError
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerTimeChanged
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerPositionChanged
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerSeekableChanged
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerPausableChanged
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerTitleChanged
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerSnapshotTaken
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerLengthChanged
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerVout
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerScrambledChanged
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerESAdded
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerESDeleted
    vlcj: (DefaultMediaPlayer.java:1784)                 | DEBUG | event=libvlc_MediaPlayerESSelected
    vlcj: (DefaultMediaPlayer.java:1731)                 | DEBUG | Media player events detached.
    vlcj: (DefaultMediaPlayer.java:1736)                 | DEBUG | Release media player...
    vlcj: (DefaultMediaPlayer.java:1738)                 | DEBUG | Media player released.
    vlcj: (DefaultMediaPlayer.java:1751)                 | DEBUG | Shut down listeners...
    vlcj: (DefaultMediaPlayer.java:1753)                 | DEBUG | Listeners shut down.
    vlcj: (MediaPlayerFactory.java:325)                  | DEBUG | release()

**Comment by caprica** [comment]
First...

LibVLC 2.2.0 provides new API, exposed through vlcj as MediaPlayer.getAudioOutputDevices().

That will give you the "friendly" strings for the devices that you are looking for (to match the VLC menus).

Second...

To set the audio output device, vlcj does this through MediaPlayer.setAudioOutputDevice().

According to the LibVLC API documentation, you can pass NULL for the audio output parameter, and one of the strings from the getAudioOutputDevices() function for the device parameter.

There is no mention in the LibVLC API documentation that this requires a restart of media playback before the change is applied (unlike changing the audio output *module* which explicitly does require a restart of playback).

As far as I know, using the ugly device strings as per the vlcj audio output device test should work, but the new API is nicer and easier.

**Comment by caprica** [comment]
In fact, I am presently working on my own advanced test case for vlcj, and I just integrated this audio device functionality.

With or without the new getAudioOutputDevices() functionality, these parameter worked for me (I have two audio devices on my computer):
```
vlcj: (DefaultMediaPlayer.java:815)                  | DEBUG | setAudioOutputDevice(output=null,outputDeviceId=alsa_output.pci-0000_01_00.1.hdmi-stereo)
vlcj: (DefaultMediaPlayer.java:815)                  | DEBUG | setAudioOutputDevice(output=null,outputDeviceId=alsa_output.pci-0000_00_1b.0.analog-stereo)
vlcj: (DefaultMediaPlayer.java:815)                  | DEBUG | setAudioOutputDevice(output=null,outputDeviceId=alsa_output.pci-0000_01_00.1.hdmi-stereo)
vlcj: (DefaultMediaPlayer.java:815)                  | DEBUG | setAudioOutputDevice(output=null,outputDeviceId=alsa_output.pci-0000_00_1b.0.analog-stereo)
```
This works, and it changes the audio output device dynamically *without* restarting the audio playback.

You can see that the device ID strings are kind of low level, and are similar to what you posted about the vlcj audio output test case, but in this instance those device ID strings came from the list of AudioDevice instances returned by MediaPlayer.getAudioOutputDevices().

So my point is, it does work - if you can use the new API then use that, if you can't use the new API you can still do it but you need to pick the right device id.

**Comment by caprica** [comment]
Unfortunately the way LibVLC works I see two issues:

1. You can't query what the current audio device is;
2. You can't change the audio device unless  the media is playing.

Those are LibVLC limitations, not vlcj limitations.

**Comment by Joe-Merten** [comment]
Thanks a lot for all of this informations.

Cause there is currently no `MediaPlayerAudioOutputDevicesTest`, I'd added the output of `MediaPlayer.getAudioOutputDevices()` to my application. 
Unfortunately it crashes on kubuntu 14.04 (both 32 and 64 bit), but on OSX I got the expected result:

    Available audio devices for this player:
      id = "0", longName = "System Sound Output Device"
      id = "38", longName = "Built-in Output"
      id = "72", longName = "BM-8810"

I think, the libvlc 2.2.0-rc2 of my OSX is some days more recent than those of my kubuntu's, so I will wait for 2.2.0 final and then try again.

**Comment by caprica** [comment]
The reason it crashes is most likely because of this VLC bug:

https://trac.videolan.org/vlc/ticket/13655

Which is now fixed in vlc-git and vlc-2.2-git.

**Comment by caprica** [comment]
By the way, with LibVLC 3.0.0 you *will* be able to query what the current device is (with some caveats).

See #304.
